tokio/src/sync/semaphore.rs

```diff
@@ -24,6 +24,51 @@ use std::sync::Arc;
 /// To use the `Semaphore` in a poll function, you can use the [`PollSemaphore`]
 /// utility.
 ///
+/// # Examples
+///
+/// ```
+/// use std::sync::Arc;
+/// use tokio::sync::{Semaphore, TryAcquireError};
+///
+/// #[tokio::main]
+/// async fn main() {
+///     let semaphore = Semaphore::new(3);
+///     {
```

Comment:
This example should probably be split into two examples. It could also use some empty lines to visually separate things.